GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,214 @@
+
+        [Option(
+            'n',
+            Required = false,
+            HelpText = "The number of directories to display hydration levels for")]
+        public int DirectoryDisplayCount { get; set; } = 5;
+
+        [Option(
+            'd',
+            "directory",
+            Required = false,
+            HelpText = "Run the statistics tool on a specific directory")]
+        public string Directory { get; set; }
+
+        protected override string VerbName
+        {
+            get { return HealthVerbName; }
+        }
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            // The path to the root of git's tree
+            string sourceRoot = enlistment.WorkingDirectoryRoot;
+
+            if (this.Directory == null)
+            {
+                this.Directory = string.Empty;
+            }
+            else if (this.Directory.Equals("."))
+            {
+                this.Directory = Environment.CurrentDirectory.Substring(sourceRoot.Length);
+            }
+
+            // Get all of the data needed to calculate statistics then pass them into GVFSEnlistmentStatistics
+            List<string> modifiedPathsFileList = new List<string>();
+            List<string> modifiedPathsFolderList = new List<string>();
+            List<string> placeholderFilePathList = new List<string>();
+            List<string> placeholderFolderPathList = new List<string>();
+
+            this.GetPlaceholdersFromDatabase(enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders);
+            foreach (IPlaceholderData placeholderData in filePlaceholders)
+            {
+                placeholderFilePathList.Add(placeholderData.Path);
+            }
+
+            foreach (IPlaceholderData placeholderData in folderPlaceholders)
+            {
+                placeholderFolderPathList.Add(placeholderData.Path);
+            }
+
+            foreach (string path in this.GetModifiedPathsFromPipe(enlistment))
+            {
+                if (path.Last() == GVFSConstants.GitPathSeparator)
+                {
+                    path.TrimEnd('/');
+                    modifiedPathsFolderList.Add(path);
+                }
+                else
+                {
+                    modifiedPathsFileList.Add(path);
+                }
+            }
+
+            List<string> gitPathsList = this.GetPathsFromGitIndex(enlistment);
+
+            GVFSEnlistmentHealthCalculator enlistmentHealthCalculator = new GVFSEnlistmentHealthCalculator(gitPathsList, placeholderFolderPathList, placeholderFilePathList, modifiedPathsFolderList, modifiedPathsFileList);
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentData = enlistmentHealthCalculator.CalculateStatistics(this.Directory, this.DirectoryDisplayCount);
+
+            string trackedFilesCountFormatted = enlistmentData.GitTrackedFilesCount.ToString("N0");

```

Comment:
What would you think of moving the "output" logic into its own method (or possibly into a `Presenter` type?). I think it would be read easier (and be be easier to update if necessary) if we put the details of how we output the results into a separate method - so at the top level this method would be something like:

```
Execute(...)
{
    /// Some preparation....
   HealthCalculator healthCalculator = new HealthCalculator(...);
   enlistmentData = healthCalculator.Calculate(...);
   Output(enlistmentData);
}
```

Where the details of `Calculate` and `Output` were contained in their own classes / methods.

Reply:
I created `PrintOutput()` for now which houses the output commands. There's probably more refactoring that could be done to smooth it out but I think that may be better left to a follow up pr